core-services/src/main/java/org/zalando/nakadi/service/SchemaService.java

```diff
@@ -314,7 +314,28 @@ public String getAvroSchemaVersion(final String name, final org.apache.avro.Sche
         );
     }
 
-    private class SchemaId {
+    @Override
+    public String getSchemaVersion(final String name, final String schema,
+                                   final EventTypeSchemaBase.Type type)
+            throws NoSuchSchemaException {
+        switch (type) {
+            case AVRO_SCHEMA:
+                return getAvroSchemaVersion(name,  new Parser().parse(schema));
+            case JSON_SCHEMA:
+                final JSONObject jsonSchema = new JSONObject(schema);
+                return schemaRepository.getSchemas(name, 0, 100).stream()
```

Comment:
It feels like here and in the avro function, the limit should not be specified — it only makes sense for external requests.

Also, if we only return one object, makes sense to ensure ordering, e.g. to return the latest version that matches the schema.

Reply:
Created a getAll schemas method that doesnt use limit & offsets, regarding giving the latest version, the db query orders by `created_at desc` and we return the first match in code, so should be the latest.